v1/plugin/plugin.go

```diff
@@ -104,16 +149,52 @@ type tlsServerSetup interface {
 }
 
 // osInputOutput supports interactions with OS for the plugin lib
-type osInputOutput interface {
+type OSInputOutput interface {
 	// readOSArgs gets command line arguments passed to application
-	readOSArgs() []string
+	readOSArg() string
 	// printOut outputs given data to application standard output
 	printOut(data string)
+
+	args() int
+
+	setContext(c *cli.Context)
 }
 
 // standardInputOutput delivers standard implementation for OS
 // interactions
 type standardInputOutput struct {
+	context *cli.Context
+}
+
+// libInputOutput holds utility used for OS interactions
+var libInputOutput OSInputOutput = &standardInputOutput{}
+
+// readOSArgs implementation that returns application args passed by OS
+func (io *standardInputOutput) readOSArg() string {
+	if io.context != nil {
+		return io.context.Args().First()
+	}
+	if len(os.Args) > 0 {
+		return os.Args[0]
+	}
+	return ""
+}
+
+// printOut implementation that emits data into standard output
+func (io *standardInputOutput) printOut(data string) {
+	fmt.Println(data)
+}
+
+func (io *standardInputOutput) setContext(c *cli.Context) {
+	io.context = c
+}
+
+//TODO(JC) where is this used?  Should this be renamed?
+func (io *standardInputOutput) args() int {
```

Comment:
Based on code, args() is used in [plugin.go#L530](https://github.com/intelsdi-x/snap-plugin-lib-go/pull/84/files#diff-01c2cbb747cc453d78a7b29d3e51c7d7R530) - is that right? If yes, "TODO" comment might be removed I suppose.